scripts/import_from_biosamples.py

```diff
@@ -302,14 +307,24 @@ def unify_field_names(biosample):
 
 
 def find_essential_fields(biosample: Dict) -> bool:
+    """
+    check whether the essential field(s) exist in the sample record
+    currently, essential field list only includes Material
+    :param biosample: sample record
+    :return: true if all essential fields are found
+    """
     essential_fields = ['Material']
     for essential in essential_fields:
         if essential not in biosample['characteristics']:
             return False
     return True
```

Comment:
Why do you need to do this check, validator already should do it for you

Reply:
many codes are based on many assumptions, e.g. the API endpoint and result structure does not change and all mandatory fields are in the data if the submitter claims to be FAANG standard (using our tools to validate and convert the template). The reality is that a few records do not meet this requirement for different reasons, e.g. BioSamples copy down pipeline does not work properly, curation goes wrong etc.